devops/ansible/roles/girder/library/girder.py

```diff
@@ -942,10 +950,15 @@ def update(self, _id, body, **kwargs):
         if _id in self.resources:
             current = self.resources[_id]
             # if body is a subset of current we don't actually need to update
-            if set(body.items()) <= set(current.items()):
-                return current
-            else:
-                return self.__apply(_id, self.client.put, body, **kwargs)
+            try:
+                if set(body.items()) <= set(
```

Comment:
This should be sufficient for the general case.  We could instead do:
```
  if all([k in current and v == current.get(k) for k, v in body.items()]):
    return current
```

There are a few other spots in the library/girder.py file where we use the `set(foo.items())` motif (such as in the `setting` method).  These are probably brittle and should be tested.

Maybe @kotfic or @danlamanna can chime in.